### PR TITLE
ARGV: Replace usages of ARGV.resolved_formulae with Homebrew.args

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
#5730

### CODE REVIEW

File: Library/Homebrew/cmd/--cellar.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def __cellar
     if Homebrew.args.named.blank?
       puts HOMEBREW_CELLAR
     else
-      puts ARGV.resolved_formulae.map(&:rack)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/--cellar.rb]: Can the `ARGV` method be `odeprecated`?

Comment (GauthamGoli) [reply Library/Homebrew/cmd/--cellar.rb]: Yes. Updating the method accordingly

Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/--cellar.rb]: Great, thanks. If that can be done for any of the other methods you've done in previous PRs that would be great, too.

Comment (GauthamGoli) [reply Library/Homebrew/cmd/--cellar.rb]: The way we run `postinstall` is a bit tricky and is blocking `ARGV#resolved_formulae` from being deprecated. 
1. [`cmd/postinstall.rb`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cmd/postinstall.rb) calls [`FormulaInstaller#post_install`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula_installer.rb#L909) which then runs [`Homebrew/postinstall.rb`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/postinstall.rb) in a sandbox

2. `Homebrew/postinstall.rb` does not use the parser but uses `ARGV.resolved_formulae` 

We could either
 - use `cli/parser` to parse args within `Homebrew/postinstall.rb`
 - Add fallback to ARGV in `CLI::Args#resolved_formulae`

@MikeMcQuaid Need your input on the above two approaches. I can attempt this in a separate PR.

Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/--cellar.rb]: > * use `cli/parser` to parse args within `Homebrew/postinstall.rb`

This seems better 👍. Thanks!

Are there any other `ARGV` methods that can be removed or `odeprecated` that were previously ported?

**Comment by GauthamGoli** [comment]
@MikeMcQuaid I am not able to reproduce the failing test locally, I am instead getting a segmentation error for that particular test. Any pointers/ideas on how do to debug it ?

**Comment by MikeMcQuaid** [comment]
@GauthamGoli What macOS version are you on?

**Comment by MikeMcQuaid** [comment]
(this is an issue that seems to happen sometimes with the portable ruby only when running `brew tests`).

**Comment by GauthamGoli** [comment]
I am on macOS High Sierra 10.13.6

File: Library/Homebrew/cmd/--cellar.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/--cellar.rb]: > Are there any other `ARGV` methods that can be removed or `odeprecated` that were previously ported?

Checked these and done them!

**Comment by MikeMcQuaid** [comment]
Actually, let's just 🚢. Thanks again @GauthamGoli!

**Pull request merged**